### ISSUE TITLE
Receiving and error while trying to connect with an existing remote jupyter notebook server

### ISSUE THREAD

**Issue by leandroohf** [microsoft/vscode-jupyter#13714]
## Environment data

-   VS Code version: 1.79.1
-   Jupyter Extension version (available under the Extensions sidebar): v2023.5.1001582324
-   Python Extension version (available under the Extensions sidebar): v2023.10.1
-   OS (Windows | Mac | Linux distro) and version: Windows 10 Enterprise LTSC version 1809
-   Python and/or Anaconda version: 3.9.2
-   Type of virtual environment used (venv ): venv
-   Jupyter server running: Remote

## Expected behaviour

After: Select Another Kernel -> Existing Jupyter Server and copy and paste the URL:  http://localhost:8888/?token=<MY_TOKEN HERE\>. I was expecting to connect to the remote Jupyter server and be able to run the opened notebook on that server. 

## Actual behaviour
Could not connect to the server. Receive an error msg on the Jupyter Output console.

## Steps to reproduce:

1. On remote machine (Linux):

jupyter  --version
jupyter core     : 4.7.1
jupyter-notebook : 6.2.0
qtconsole        : 5.4.0
ipython          : 7.20.0
ipykernel        : 5.4.3
jupyter client   : 7.0.0
jupyter lab      : not installed
nbconvert        : 7.0.0
ipywidgets       : 6.0.0
nbformat         : 5.1.2
traitlets        : 5.7.1

jupyter-notebook --no-browser --port=8888 --allow-root --ip="\*"  --NotebookApp.allow_origin='\*'

2.  local machine WSL1

gcloud compute start-iap-tunnel vm-on-gcp 8888 --local-host-port=localhost:8888 --zone=us-central1-a

Test connection in the web browser. It works, and I can run the notebooks.

3. vs code in local machine 

3.1) Select Another Kernel -> Existing Jupyter Server
3.2) Paste the URI: http://localhost:8888/?token=<MY_TOKEN HERE> (same URL tested in web browser)
3.3) Get the message:  [error] Failed to parse remote URI http://localhost:8888/?token=<MY_TOKEN HERE>[Error: Invalid remote URI

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
Visual Studio Code (1.79.1, undefined, desktop)
Jupyter Extension Version: 2023.5.1001582324.
Python Extension Version: 2023.10.1.
Platform: win32 (x64).
Workspace folder ~\Downloads\algopricing-analytics, Home = c:\Users\<MY USER>
12:08:58.255 [info] Start refreshing Interpreter Kernel Picker (1686856138255)
12:08:58.265 [info] Using Pylance
12:08:59.028 [error] Failed to parse remote URI http://localhost:8888/?token=<MY_TOKEN HERE> [Error: Invalid remote URI
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:2:1694598)
	at hk.parseUri (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:24:529924)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:24:527677]
```

</p>
</details>

**Comment by hhessel** [comment]
I've got same problem on both my Mac and Windows VSCode installation.

Visual Studio Code (1.79.2, undefined, desktop)
Jupyter Extension Version: 2023.5.1001582324.
Python Extension Version: 2023.10.1.
```
10:29:51.262 [error] Failed to setup finder for Select Kernel [Error: 'Failed to parse remote URI https://lab.hph-network.de/lab?token=xxxx
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:2:1694682)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:17:126648
	at async Promise.all (index 8)
	at async c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:17:126492]
```

**Comment by DonJayamanne** [comment]
please could you install vscode insiders and the pre release version of Jupyter extension and let me know if the issue persists in there as well

* Please install latest VS Code Insiders
* Open the extensions page, search for Jupyter and click the button `Switch to Pre-Release Version`
* Similarly please install the latest `Pre-Release` version of the Python extension

<img width="572" alt="Screenshot 2023-02-04 at 09 24 37" src="https://user-images.githubusercontent.com/1948812/216722719-fd8e14dc-783f-4859-8235-012b1bc796d9.png">

**Comment by hhessel** [comment]
Hi @DonJayamanne 

The kernel picker shows up in the insiders version, but interestingly enough a similar error as above is still thrown in the jupyter outlog panel.

![image](https://github.com/microsoft/vscode-jupyter/assets/225331/99ab4793-823f-4d06-bd65-9c9d3c942754)

```
11:23:18.892 [error] Failed to parse remote URI https://lab.hph-network.de/lab?token=xxx[Error: Invalid remote URI
	at $r (c:\Users\<username>\.vscode-insiders\extensions\ms-toolsai.jupyter-2023.6.1001681100-win32-x64\out\extension.node.js:2:1693671)
	at Pk.parseUri (c:\Users\<username>\.vscode-insiders\extensions\ms-toolsai.jupyter-2023.6.1001681100-win32-x64\out\extension.node.js:24:531214)
	at c:\Users\<username>\.vscode-insiders\extensions\ms-toolsai.jupyter-2023.6.1001681100-win32-x64\out\extension.node.js:24:529944
	at p.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:134)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2051)
	at i.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1692)
	at h._fireDidAccept (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:101:110331)
	at v.$onDidAccept (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:101:108576)
	at n.N (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:11196)
	at n.M (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:10962)
	at n.H (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:10043)
	at n.G (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:9119)
	at c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:7949
	at p.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:134)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2051)
	at i.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1692)
	at w.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:73:13827)
	at c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:127:30350
	at p.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:134)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2051)
	at i.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1692)
	at w.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:73:13827)
	at MessagePortMain.<anonymous> (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:127:28630)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)]
```

**Comment by DonJayamanne** [comment]
Ok I’m a little confused
Let’s ignore the error message, can you describe what is not working
Do you get an error message when you select a kernel, if  so please can you share a screenshot of the error displayed
Or share a screen recording of the problem

**Comment by DonJayamanne** [comment]
Or is the problem the fact that no kernels are displayed in the kernel picker in the stable release version,  
But works in the pre-release version (ignoring the error in the logs)

**Comment by hhessel** [comment]
Hi @DonJayamanne 

No problem, I'm appreciate the time you take to look into this.

As soon as I enter the remote (Step 1/2) Jupyter Server URL I receive the following error in the Jupyter Output Panel

```
12:03:46.200 [error] Failed to parse remote URI https://lab.hph-network.de/lab?token=xxx [Error: Invalid remote URI
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:2:1694598)
	at hk.parseUri (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:24:529924)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:24:528712
	at b.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2121)
	at n.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1729)
	at f._fireDidAccept (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:101:114142)
	at E.$onDidAccept (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:101:112352)
	at s.N (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:11717)
	at s.M (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:11435)
	at s.H (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:10516)
	at s.G (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:9494)
	at c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:8282
	at b.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2121)
	at n.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1729)
	at p.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:73:14783)
	at c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:127:32461
	at b.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2121)
	at n.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1729)
	at p.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:73:14783)
	at MessagePortMain.<anonymous> (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:127:30587)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)]
```

No Error on Step (2/3) Change Display Name

On Step (3/3) Select Kernel it just loads indefinitely and display the following error in the output log.

![image](https://github.com/microsoft/vscode-jupyter/assets/225331/10a17e86-822d-4b6d-ba43-2927f03e7459)

```
12:05:55.320 [error] Failed to setup finder for Select Kernel [Error: 'Failed to parse remote URI https://lab.hph-network.de/lab?token=xxx
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:2:1694682)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:17:126648
	at async Promise.all (index 8)
	at async c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1001582324-win32-x64\out\extension.node.js:17:126492]

```

Looks a bit like a downstream electron error maybe?

**Comment by DonJayamanne** [comment]
@hhessel 
Does the same thing work in vscode insiders and the pre release version?

**Comment by hhessel** [comment]
> @hhessel Does the same thing work in vscode insiders and the pre release version?

Yes, it does work flawlessly as shown in comment https://github.com/microsoft/vscode-jupyter/issues/13714#issuecomment-1594391866 above.

**Comment by DonJayamanne** [comment]
@hhessel aah yes, very sorry I missed that

Please could you install this VSIX into your stable version of VS Code and let me know if this works

* Download the file
* Extract the file
* Open VS Code and select the command `Install from VSIX`
* Select the above extracted VSIX
* Reload VS Code and check if it works
[ms-toolsai-jupyter-insiders.vsix.zip](https://github.com/microsoft/vscode-jupyter/files/11769086/ms-toolsai-jupyter-insiders.vsix.zip)

**Comment by ofhtech** [comment]
I can confirm that the current Jupyter extension release version (v2023.5.1001582324) works on

Version: 1.80.0-insider (user setup)
Commit: 73eb619afb87cab4aa83a569259c891b6983c11e
Date: 2023-06-19T05:35:37.388Z
Electron: 22.3.11
ElectronBuildId: 21658788
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

but is giving me the errors described in the OP post in the following vscode version:

Version: 1.79.2 (user setup)
Commit: 695af097c7bd098fbf017ce3ac85e09bbc5dda06
Date: 2023-06-14T08:57:04.379Z
Electron: 22.5.7
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

**Comment by leandroohf** [comment]
I still have the issue. See logs below:

Visual Studio Code - Insiders (1.80.0-insider, wsl, desktop)
Jupyter Extension Version: 2023.6.1001691100.
Python Extension Version: 2023.10.1.
Platform: linux (x64).
Workspace folder ~/Documents/leandro-algopricing-analytics, Home = /home/leandroohf
12:42:49.541 [info] User belongs to experiment group 'FastKernelPicker'
12:42:49.541 [info] User belongs to experiment group 'PasswordManager'
12:42:49.814 [info] Start refreshing Kernel Picker (1687207369814)
12:42:49.833 [info] Using Pylance
12:42:50.334 [warn] No interpreter for Pylance for Notebook URI "~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb"
12:42:50.876 [info] End refreshing Kernel Picker (1687207369814)
12:43:04.847 [info] Start refreshing Interpreter Kernel Picker
12:43:04.847 [info] Start refreshing Kernel Picker (1687207384847)
12:43:05.700 [warn] No interpreter with path ~/algo-pricing/algo-env3.6/bin/python found in Python API, will convert Uri path to string as Id ~/algo-pricing/algo-env3.6/bin/python
12:43:05.705 [warn] No interpreter with path ~/.venvs/rb-pipeline/bin/python found in Python API, will convert Uri path to string as Id ~/.venvs/rb-pipeline/bin/python
12:43:14.735 [info] End refreshing Kernel Picker (1687207384847)
12:45:26.211 [error] Failed to parse remote URI http:://localhost:8888 [Error: Invalid remote URI
	at $r (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001691100-linux-x64/out/extension.node.js:2:1693667)
	at Ok.parseUri (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001691100-linux-x64/out/extension.node.js:24:549565)
	at ~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001691100-linux-x64/out/extension.node.js:24:548054
	at p.invoke (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at h._fireDidAccept (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:102:110322)
	at v.$onDidAccept (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:102:108567)
	at n.N (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:112:11196)
	at n.M (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:112:10962)
	at n.H (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:112:10043)
	at n.G (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:112:9119)
	at ~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:112:7949
	at p.invoke (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at w.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:76:13827)
	at ~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:128:30351
	at p.invoke (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at w.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:76:13827)
	at s.z (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:76:16872)
	at ~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:76:15324
	at p.invoke (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at i.acceptChunk (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:76:11601)
	at ~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:76:10888
	at p.invoke (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at v.u (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:128:19085)
	at v.acceptFrame (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:128:18891)
	at d.l (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:128:17524)
	at ~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:128:14871
	at Socket.r (~/.vscode-server-insiders/bin/73eb619afb87cab4aa83a569259c891b6983c11e/out/vs/workbench/api/node/extensionHostProcess.js:128:12785)
	at Socket.emit (node:events:513:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Socket.Readable.push (node:internal/streams/readable:228:10)
	at TCP.onStreamRead (node:internal/stream_base_commons:190:23)]

**Comment by yuerlong** [comment]
I had a similar issue when jupyter notebook is connected to sparkmagic after 1.74. currently, I am keeping using Jupyter extension v2023.1.1003441034. It seems that after v2023.1.1003441034, there are always some issues.

**Comment by leandroohf** [comment]
Because of https://github.com/microsoft/vscode-jupyter/issues/13714#issuecomment-1597422040. I realized [ofhtech](https://github.com/ofhtech) is using a different Jupyter extension version than me. I managed to switch to that version. 
And after throwing the same error messages I mentioned before ([error] Failed to parse remote URI), the instructions worked for me. I could connect to the remote Jupyter Notebook server and run the first cells in the remote machine.

I am adding the logs here in case they might be useful. 

Visual Studio Code - Insiders (1.80.0-insider, wsl, desktop)
Jupyter Extension Version: 2023.5.1001582324.
Python Extension Version: 2023.10.1.
Platform: linux (x64).
Workspace folder ~/Documents/leandro-algopricing-analytics, Home = /home/leandroohf
15:10:19.769 [info] Start refreshing Interpreter Kernel Picker (1687216219769)
15:10:19.815 [info] Using Pylance
15:10:20.743 [error] Failed to parse remote URI http://localhost:8888/?token=f82c885b745cc922ab8c21f153ed9bd6985c530d22b0cefb [Error: Invalid remote URI
	at qr (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:2:1694598)
	at hk.parseUri (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:24:529924)
	at ~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:24:527677
	at processTicksAndRejections (node:internal/process/task_queues:96:5)]
15:10:24.565 [info] Dispose Kernel '~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb' associated with '~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb'
15:10:24.594 [warn] Failed to get activated env vars for /bin/python3 in 1680ms
15:10:24.924 [info] Process Execution: /bin/python3 -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
15:10:25.436 [info] Starting Kernel connectToLiveRemoteKernel, 38f00f94-17f7-4bff-a7ad-cf7c909770f5  for '~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb' (disableUI=true)
15:10:25.485 [error] Unable to determine site packages path for python /bin/python3 (Unknown)
15:10:25.941 [info] Process Execution: /bin/python3 -m pip list
15:10:29.159 [info] Started new session 38f00f94-17f7-4bff-a7ad-cf7c909770f5
15:10:29.160 [info] Started session for kernel connectToLiveRemoteKernel:38f00f94-17f7-4bff-a7ad-cf7c909770f5
15:10:31.470 [warn] No interpreter with path ~/algo-pricing/algo-env3.6/bin/python found in Python API, will convert Uri path to string as Id ~/algo-pricing/algo-env3.6/bin/python
15:10:33.023 [info] End refreshing Interpreter Kernel Picker (1687216219769)
15:10:52.501 [info] Handle Execution of Cells 2 for ~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb
15:10:52.618 [info] Kernel acknowledged execution of cell 2 @ 1687216251516
15:10:53.170 [info] End cell 2 execution @ 1687216252064, started @ 1687216251516, elapsed time = 0.548s

**Comment by ofhtech** [comment]
I just downgraded the jupyter extension to the previous release v2023.4.1011241018 and it is working again for me, so it appears it's an issue with the latest version (v2023.5.1001582324) of the extension, using vscode 1.79.2 at least for me.

**Comment by DonJayamanne** [comment]
I'm sorry you are running into this issue, please could you try the following:
* Downgrade to the old extension then use the command `Jupyter: Clear Jupyter Remote Server List`
* Upgrade to the new version of the extesnion
* Add the new URL once again and let me know if thsi works

@leandroohf @ofhtech Sorry you are running into this issue, please could you try the above and let me know if the new 2023.5.* version works as expected

**Comment by DonJayamanne** [comment]
Duplicate of https://github.com/microsoft/vscode-jupyter/issues/13699

**Comment by leandroohf** [comment]
Well, did not work for me. I still have this error.

Visual Studio Code - Insiders (1.80.0-insider, wsl, desktop)
Jupyter Extension Version: 2023.5.1001582324.
Python Extension Version: 2023.10.1.
Platform: linux (x64).

10:27:47.423 [error] Failed to parse remote URI http://localhost:8888/?token=0cbe88ef2083abc34e01142a46dbcf560d55bd495467bc25 [Error: Invalid remote URI

But now, I cannot connect to remote jupyter server neither run any cell in vs code insiders.

**Comment by DonJayamanne** [comment]
> cannot connect to remote jupyter server anymore.

please install the latest pre-release version
2023.6.*

> But now, I cannot connect to remote jupyter server anymore.

install the latest pre release version, clear the url list  and enter the url again

**Comment by leandroohf** [comment]
Still having issues.

Visual Studio Code - Insiders (1.80.0-insider, wsl, desktop)
Jupyter Extension Version: 2023.6.1001731100.
Python Extension Version: 2023.10.1.
Platform: linux (x64).
Workspace folder ~/Documents/leandro-algopricing-analytics, Home = /home/leandroohf
15:40:30.476 [info] User belongs to experiment group 'FastKernelPicker'
15:40:30.477 [info] User belongs to experiment group 'PasswordManager'
15:40:30.777 [info] Start refreshing Kernel Picker (1687304430777)
15:40:30.798 [info] Using Pylance
15:40:32.270 [warn] No interpreter with path ~/algo-pricing/algo-env3.6/bin/python found in Python API, will convert Uri path to string as Id ~/algo-pricing/algo-env3.6/bin/python
15:40:36.063 [warn] Failed to get activated env vars for /bin/python3 in 3762ms
15:40:36.381 [info] Process Execution: /bin/python3 -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
15:40:36.494 [error] Unable to determine site packages path for python /bin/python3 (Unknown)
15:40:36.775 [info] Process Execution: /bin/python3 -m pip list
15:40:36.877 [warn] No interpreter for Pylance for Notebook URI "~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb"
15:40:37.456 [info] End refreshing Kernel Picker (1687304430777)
15:40:39.369 [error] Failed to parse remote URI ttp:://localhost:8888 [Error: Invalid remote URI
	at $r (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:2:1693667)
	at Ok.parseUri (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:549591)
	at ~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:548080
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at h._fireDidAccept (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:102:110322)
	at v.$onDidAccept (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:102:108567)
	at n.N (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:11196)
	at n.M (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:10962)
	at n.H (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:10043)
	at n.G (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:9119)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:7949
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at w.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:13827)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:30351
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at w.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:13827)
	at s.z (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:16872)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:15324
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at i.acceptChunk (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:11601)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:10888
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at v.u (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:19085)
	at v.acceptFrame (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:18891)
	at d.l (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:17524)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:14871
	at Socket.r (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:12785)
	at Socket.emit (node:events:513:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Socket.Readable.push (node:internal/streams/readable:228:10)
	at TCP.onStreamRead (node:internal/stream_base_commons:190:23)]
15:40:39.374 [error] Failed to get the password for ttp://localhost:8888/ [TypeError: Only HTTP(S) protocols are supported
	at ~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:2:1138431
	at ~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:2:1139127
	at new Promise (<anonymous>)
	at Y (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:2:1138162)
	at xk.makeRequest (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:541233)
	at xk.isJupyterHub (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:541856)
	at xk.getNonCachedPasswordConnectionInfo (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:536262)
	at xk.getPasswordConnectionInfo (~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:535933)
	at ~/.vscode-server-insiders/extensions/ms-toolsai.jupyter-2023.6.1001731100-linux-x64/out/extension.node.js:24:548215
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at h._fireDidAccept (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:102:110322)
	at v.$onDidAccept (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:102:108567)
	at n.N (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:11196)
	at n.M (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:10962)
	at n.H (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:10043)
	at n.G (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:9119)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:112:7949
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at w.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:13827)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:30351
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at w.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:13827)
	at s.z (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:16872)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:15324
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at i.acceptChunk (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:11601)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:76:10888
	at p.invoke (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:134)
	at deliver (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:2051)
	at i.fire (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:60:1692)
	at v.u (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:19085)
	at v.acceptFrame (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:18891)
	at d.l (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:17524)
	at ~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:14871
	at Socket.r (~/.vscode-server-insiders/bin/8c85702bca51b08005ada7fc4eaf52a61623b2b6/out/vs/workbench/api/node/extensionHostProcess.js:128:12785)
	at Socket.emit (node:events:513:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Socket.Readable.push (node:internal/streams/readable:228:10)
	at TCP.onStreamRead (node:internal/stream_base_commons:190:23)]

**Comment by DonJayamanne** [comment]
@leandroohf 
* When do you run into this issue, is this when you are trying to enter the new URL into the prompt
* Or is this when you load VS Code 
* Or is this when you attempt to run a cell?

> 15:40:37.456 [info] End refreshing Kernel Picker (1687304430777)
15:40:39.369 [error] Failed to parse remote URI ttp:://localhost:8888 [Error: Invalid remote URI

Looks like you have missed the starting `h` in the URL (see error `ttp:://localhost:8888`)
Please can you confirm you have pasted the full URL

**Comment by DonJayamanne** [comment]
I'll make a change to ensure we validate this in the input itself and ensure you do not run into this again.
Let me know if the above works (i.e. entering the URL with the missing `h`, if thats the case)

**Comment by leandroohf** [comment]
@DonJayamanne I might have missed the `h` in the URL for the first time. This time a double check the URL.

`http://localhost:8888/?token=0cbe88ef2083abc34e01142a46dbcf560d55bd495467bc25`

See logs: 

```
Visual Studio Code - Insiders (1.80.0-insider, wsl, desktop)
Jupyter Extension Version: 2023.6.1001731100.
Python Extension Version: 2023.10.1.
Platform: linux (x64).
Workspace folder ~/Documents/leandro-algopricing-analytics, Home = /home/leandroohf
20:27:55.547 [info] User belongs to experiment group 'FastKernelPicker'
20:27:55.547 [info] User belongs to experiment group 'PasswordManager'
20:27:56.225 [info] Start refreshing Kernel Picker (1687321676225)
20:27:56.280 [info] Using Pylance
20:27:59.231 [warn] No interpreter with path ~/algo-pricing/algo-env3.6/bin/python found in Python API, will convert Uri path to string as Id ~/algo-pricing/algo-env3.6/bin/python
20:28:02.792 [warn] Failed to get activated env vars for /bin/python3 in 3505ms
20:28:03.110 [info] Process Execution: /bin/python3 -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
20:28:03.184 [error] Unable to determine site packages path for python /bin/python3 (Unknown)
20:28:03.462 [info] Process Execution: /bin/python3 -m pip list
20:28:03.850 [warn] No interpreter for Pylance for Notebook URI "~/Documents/leandro-algopricing-analytics/dev/test_implement_rouse_algo-pricing/2023-06-11-lhof-ee_learning_curve.ipynb"
20:28:04.970 [info] End refreshing Kernel Picker (1687321676225)
20:28:28.317 [error] Failed to parse remote URI http:://localhost:8888 [Error: Invalid remote URI
```

**Comment by JamesCoveney** [comment]
> I just downgraded the jupyter extension to the previous release v2023.4.1011241018 and it is working again for me, so it appears it's an issue with the latest version (v2023.5.1001582324) of the extension, using vscode 1.79.2 at least for me.

worked for me as well, as I have been receiving the same errors since the latest release of the Jupyter extension
